packages/sdk/src/messageExtension/executeWithSSO.ts

```diff
@@ -0,0 +1,125 @@
+import { parseJwt, getScopesArray, formatString } from "../util/utils";
+import { TeamsMsgExtTokenResponse } from "./teamsMsgExtTokenResponse";
+import { ErrorWithCode, ErrorCode, ErrorMessage } from "../core/errors";
+import { AuthenticationConfiguration } from "../models/configuration";
+import { internalLogger } from "../util/logger";
+import { TeamsFx } from "../core/teamsfx";
+/**
+ * Retrieve the OAuth Sign in Link to use in the MessagingExtensionResult Suggested Actions.
+ * This method only work on MessageExtension with Query now.
+ *
+ * @param {TeamsFx} teamsfx - Used to provide configuration and auth.
+ * @param {string | string[]} scopes - The list of scopes for which the token will have access.
+ *
+ * @returns SignIn link CardAction with 200 status code.
+ */
+function getSignInResponseForMessageExtension(teamsfx: TeamsFx, scopes: string | string[]): any {
+  const scopesArray = getScopesArray(scopes);
+  const signInLink = `${teamsfx.getConfig("initiateLoginEndpoint")}?scope=${encodeURI(
+    scopesArray.join(" ")
+  )}&clientId=${teamsfx.getConfig("clientId")}&tenantId=${teamsfx.getConfig("tenantId")}`;
+  return {
+    composeExtension: {
+      type: "silentAuth",
+      suggestedActions: {
+        actions: [
+          {
+            type: "openUrl",
+            value: signInLink,
+            title: "Message Extension OAuth",
+          },
+        ],
+      },
+    },
+  };
+}
+
+/**
+ * execution in message extension with SSO token.
+ *
+ * @param {TurnContext} context - The context object for the current turn.
+ * @param {AuthenticationConfiguration} config - User custom the message extension authentication configuration.
+ * @param {string[]} scopes - The list of scopes for which the token will have access.
+ * @param {function} logic - The user execution code with SSO token.
+ *
+ * @throws {@link ErrorCode|InternalError} when failed to get access token with unknown error.
+ * @throws {@link ErrorCode|TokenExpiredError} when SSO token has already expired.
+ * @throws {@link ErrorCode|ServiceError} when failed to get access token from simple auth server.
+ * @throws {@link ErrorCode|InvalidParameter} when scopes is not a valid string or string array.
+ * @throws {@link ErrorCode|RuntimeNotSupported} when runtime is nodeJS.
+ *
+ * @returns A MessageExtension Response for the activity. If the logic not return any, return void instead.
+ */
+export async function executionWithToken(
+  context: TurnContext,
+  config: AuthenticationConfiguration,
+  scopes: string | string[],
+  logic?: (token: TeamsMsgExtTokenResponse) => Promise<any>
+): Promise<MessagingExtensionResponse | void> {
+  const valueObj = context.activity.value;
+  if (!valueObj.authentication || !valueObj.authentication.token) {
+    internalLogger.verbose("No AccessToken in request, return silentAuth for AccessToken");
+    return getSignInResponseForMessageExtension(new TeamsFx(undefined, config), scopes);
```

Comment:
Maybe use `new TeamsFx(IdentityType.User, config)` can help user better understand that we're using user delegate permission?

Reply:
sure, thanks